README.md

```diff
@@ -27,103 +32,103 @@ Then, build the project with:
 
-### Generating an example
+**voodoo** is run in three successives steps detailed below.
 
-To create an example, run:
+## :seedling: 1. Prepare the packages: voodoo-prep
 
-```bash
-make example
+To prepare the packages just run:
+
+```sh
+voodoo-prep [-u <UNIVERSES>]
 ```
 
-This will create a sample website with the documentation of `ocaml-base-compiler` in `_generated/output/p`.
+Where `<UNIVERSES>` is a comma-separated list of packages and universes `pkg1:unv1,pkg2:unv2,...`.
 
-Here are some commands that will be executed when running the above:
+Each package installed in the current switch will be prepared (prepped) for the compiling step.
 
-```bash
-$ mkdir _generated
-$ cd _generated; opam exec -- dune exec -- voodoo-prep
-$ cd _generated; opam exec -- dune exec -- voodoo-do -p ocaml-base-compiler -b
-$ cd _generated; opam exec -- dune exec -- voodoo-gen -o output
-```
+This creates an ad-hoc directory structure and populates it with `.cmt`, `.cmi`, `.cmti`, `.mld` from the current switch.
+The `ocamlobjinfo` of the `.cma` files are also copied, as well as the documentation files such as `.md`, `.html` and others (such as the `opam` files). 
 
-Read further for the detailed version of the above commands.
 
-```bash
-$ cd _generated
-$ opam exec -- dune exec -- voodoo-prep
-Warning: No universes have been specified: will generate dummy universes
-$ opam exec -- dune exec -- voodoo-do -p ocaml-base-compiler -b
-$ opam exec -- dune exec -- voodoo-do -p result -b
-$ opam exec -- dune exec -- voodoo-do ...
+## :sweat_drops: 2. Compiling and linking the packages: voodoo-do
+
+Once `voodoo-prep` has run, `voodoo-do` is able to process packages. Simply run:
+
+```sh
+voodoo-do -p <PACKAGE> [-b] [--failed]
 ```
 
-When new packages are installed in the current switch `voodoo-prep` needs to be executed in
-for `voodoo-do` to be able to process them. If the packages are done out of dependency order,
- `voodoo-prep` will alert that there are missing dependencies:
+This command runs `odoc compile` and `odoc link` on the package specified after `-p`.
+If the `--failed` flag is set, a file named `failed` containing `"failed"` is also generated.
 
-```bash
-$ opam exec -- dune exec -- voodoo-do -p odoc -b
+If the packages are processed out of dependency order, `voodoo-do` will alert that there are missing dependencies.
+
+```sh
+voodoo-do -p odoc -b
 ...
 Missing dependency: Stdlib c21c5d26416461b543321872a551ea0d
 ...
 ```
 
 In this case, we need to run `voodoo-do -p ocaml-base-compiler -b` first.
 
-Note that when being used in this mode, the `-b` (blessed) switch should
-always be passed to `voodoo-do`.
+Note that when being used in this mode, the `-b` flag should always be passed to `voodoo-do`.
 
 At this point, to view the output, use `odoc` to generate the support files (mostly: copying `highlight.js` from https://highlightjs.org and `odoc.css`):
 
 ```bash
-$ cd ..
-$ odoc support-files -o _generated/html
+cd ..
+odoc support-files -o html
```

Comment:
This is probably no longer relevant after `--as-json` was introduced. You cannot view the docs as HTML unless you run `odoc html` without `--as-json`.

However, what might be useful is that the docs can be viewed by running ocaml/ocaml.org locally with `OCAMLORG_DOC_URL=http://localhost:????/ make start` when they are being served by a local http server. But ocaml.org itself uses port `8080` by default.

Reply:
I manage to view the docs if I run `dream-serve` alone.

But running ocaml.org while serving the docs doesn't work for me, are there some additional steps missing?
I'm running:
```
dream-serve -p 9090 ./html
```
and
```
OCAMLORG_DOC_URL=http://localhost:9090/html make start #I tried with and without /html
```

Reply:
I was wrong: apparently, voodoo does generate the HTML docs as well - at least when running locally? So this can stay here, for local inspection of the `odoc html` output.

For OCaml.org, we instead serve the `output` folder, because that's where voodoo places the actual `odoc html --as-json` output. Like this:

```
dream-serve -p 9090 ./output
```
and
```
OCAMLORG_DOC_URL=http://localhost:9090/ make start
```